alacritty/src/daemon.rs

```diff
@@ -58,18 +59,21 @@ where
 {
     let mut command = Command::new(program);
     command.args(args).stdin(Stdio::null()).stdout(Stdio::null()).stderr(Stdio::null());
-    if let Ok(cwd) = foreground_process_path(master_fd, shell_pid) {
-        command.current_dir(cwd);
-    }
+
     unsafe {
         command
-            .pre_exec(|| {
+            .pre_exec(move || {
                 match libc::fork() {
                     -1 => return Err(io::Error::last_os_error()),
                     0 => (),
                     _ => libc::_exit(0),
                 }
 
+                // Don't fail spawning when we can't change the working directory.
```

Comment:
I think we could better describe what we do here, since the comment doesn't mention the actual purpose of this at all.

```suggestion
                // Copy foreground process' working directory, ignoring invalid paths.
```